src/st_sbardef.c

```diff
@@ -0,0 +1,332 @@
+                if (!JS_IsString(patch))
+                {
+                    return false;
+                }
+                out->patch = CachePatchName(JS_GetString(patch));
+            }
+            break;
+
+        case sbe_animation:
+            {
+                json_t *js_frames = JS_GetObject(json, "frames");
+                json_t *js_frame = NULL;
+                JS_ArrayForEach(js_frame, js_frames)
+                {
+                    sbarframe_t frame = {0};
+                    if (ParseSbarFrame(js_frame, &frame))
+                    {
+                        array_push(out->frames, frame);
+                    }
+                }
+            }
+            break;
+
+        case sbe_face:
+            out->lastattackdown = -1;
+            out->oldhealth = -1;
+            break;
+
+        case sbe_number:
+        case sbe_percent:
+            {
+                json_t *font = JS_GetObject(json, "font");
+                if (!JS_IsString(font))
+                {
+                    return false;
+                }
+                out->fontname = JS_GetString(font);
+
+                json_t *numbertype = JS_GetObject(json, "type");
+                json_t *param = JS_GetObject(json, "param");
+                json_t *maxlength = JS_GetObject(json, "maxlength");
+                if (!JS_IsNumber(numbertype) || !JS_IsNumber(param)
+                    || !JS_IsNumber(maxlength))
+                {
+                    return false;
+                }
+                out->numtype = JS_GetInteger(numbertype);
+                out->numparam = JS_GetInteger(param);
+                out->maxlength = JS_GetInteger(maxlength);
+            }
+            break;
+
+        default:
+            break;
+    }
+    return true;
+}
+
+static const char *sbe_names[] =
+{
+    [sbe_canvas] = "canvas",
+    [sbe_graphic] = "graphic",
+    [sbe_animation] = "animation",
+    [sbe_face] = "face",
+    [sbe_facebackground] = "facebackground",
+    [sbe_number] = "number",
+    [sbe_percent] = "percent"
+};
+
+static boolean ParseSbarElem(json_t *json, sbarelem_t *out)
+{
+    for (sbarelementtype_t type = sbe_none + 1; type < sbe_max; ++type)
+    {
+        json_t *obj = JS_GetObject(json, sbe_names[type]);
+        if (obj && JS_IsObject(obj))
+        {
+            return ParseSbarElemType(obj, type, out);
+        }
+    }
+
+    return false;
+}
+
+static boolean ParseNumberFont(json_t *json, numberfont_t *out)
+{
+    json_t *name = JS_GetObject(json, "name");
+    json_t *stem = JS_GetObject(json, "stem");
+    if (!JS_IsString(name) || !JS_IsString(stem))
+    {
+        return false;
+    }
+    out->name = JS_GetString(name);
+    out->stem = JS_GetString(stem);
+
+    json_t *type = JS_GetObject(json, "type");
+    if (!JS_IsNumber(type))
+    {
+        return false;
+    }
+    out->type = JS_GetInteger(type);
+
+    char lump[9] = {0};
+    boolean found = false;
```

Comment:
This needs to be an `int`.

Reply:
Fixed in https://github.com/fabiangreffrath/woof/pull/1916/commits/77ae8d3df3be9c2d97b78eaef09ff7cdf4ba7af3